numba/np/arrayobj.py

```diff
@@ -745,15 +751,53 @@ class IntegerArrayIndexer(Indexer):
     def __init__(self, context, builder, idxty, idxary, size):
         self.context = context
         self.builder = builder
-        self.idxty = idxty
-        self.idxary = idxary
+
+        self.idx_shape = cgutils.unpack_tuple(builder, idxary.shape)
         self.size = size
-        assert idxty.ndim == 1
         self.ll_intp = self.context.get_value_type(types.intp)
 
+        if idxty.ndim > 1:
+            if not context.enable_nrt:
+                raise NotImplementedError("This type of indexing is not"
+                                          " currently supported for"
+                                          " given compiler target.")
```

Comment:
Ideally this would be tested, but I'm inclined to leave this to a subsequent PR as I am relatively sure it's going to need to be moved.